server/sonar-server/src/main/java/org/sonar/server/platform/db/CheckDatabaseCharsetAtStartup.java

```diff
@@ -41,19 +39,18 @@ public CheckDatabaseCharsetAtStartup(ServerUpgradeStatus upgradeStatus, Database
 
   @Override
   public void start() {
-    check();
+    DatabaseCharsetChecker.State state = DatabaseCharsetChecker.State.STARTUP;
```

Comment:
create a function and get rid of multiple assignments to variable + add a comment